realm/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -960,4 +978,25 @@ private void fileCopy(File src, File dst) throws IOException {
         inStream.close();
         outStream.close();
     }
+
+    public void testWriteEncryptedCopy() throws Exception {
+        populateTestRealm();
+        long before = testRealm.where(AllTypes.class).count();
+        assertEquals(TEST_DATA_SIZE, before);
+
+        File destination = new File(getContext().getFilesDir(), ENCRYPTED_REALM_FILE_NAME);
+        byte[] key = new byte[64];
+        new Random(42).nextBytes(key);
+        try {
+            testRealm.writeEncryptedCopyTo(destination, key);
+        } catch(Exception e) {
+            fail();
+        }
+
+        Realm encryptedRealm = Realm.getInstance(getContext(), ENCRYPTED_REALM_FILE_NAME, key);
+        long after = encryptedRealm.where(AllTypes.class).count();
+        assertEquals(TEST_DATA_SIZE, after);
+        encryptedRealm.close();
+    }
```

Comment:
Add a test to verify that you can write the new file with a new key. Or from encrypted to non-encrypted.